### PR TITLE
Modified sample async client to be multi-threaded  

### PR DESCRIPTION
**Pull request by myeh**
A reader thread is spawned to handle all the RPC results.  IMO, this is a more typical scenario when people think of an async example.

### CODE REVIEW

**Comment by nicolasnoble** [comment]
Async and non-blocking are two different things, really, and it's important to show how the single-thread version can work. But there is value in showing your method as well I guess. How about adding a new file, showing two examples, instead of only one ?

**Comment by nicolasnoble** [comment]
LGTM, adding someone else for the second LGTM.

File: examples/cpp/helloworld/greeter_async_client2.cc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+using helloworld::HelloRequest;
+using helloworld::HelloReply;
+using helloworld::Greeter;
+
+class GreeterClient {
+  public:
+    explicit GreeterClient(std::shared_ptr<Channel> channel)
+            : stub_(Greeter::NewStub(channel)) {}
+
+    // Assambles the client's payload, sends it and presents the response back
```
Comment (vjpai) [inline examples/cpp/helloworld/greeter_async_client2.cc]: spelling. Assembles

File: examples/cpp/helloworld/greeter_async_client2.cc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    // from the server.
+    void SayHello(const std::string& user) {
+        // Data we are sending to the server.
+        HelloRequest request;
+        request.set_name(user);
+
+        AsyncClientCall* call = new AsyncClientCall;
+
+        // stub_->AsyncSayHello() perform the RPC call, returning an instance we
```
Comment (vjpai) [inline examples/cpp/helloworld/greeter_async_client2.cc]: performs

File: examples/cpp/helloworld/greeter_async_client2.cc [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        // store in "rpc". Because we are using the asynchronous API, we need the
```
Comment (vjpai) [inline examples/cpp/helloworld/greeter_async_client2.cc]: we need to

**Comment by vjpai** [comment]
have started making comments; more coming soon.

File: examples/cpp/helloworld/greeter_async_client.cc [hunk 4]
```diff
@@ -74,9 +74,9 @@ class GreeterClient {
     // Storage for the status of the RPC upon completion.
     Status status;
 
-    // stub_->AsyncSayHello() perform the RPC call, returning an instance we
-    // store in "rpc". Because we are using the asynchronous API, we need the
-    // hold on to the "rpc" instance in order to get updates on the ongoig RPC.
+    // stub_->AsyncSayHello() performs the RPC call, returning an instance we
+    // store in "rpc". Because we are using the asynchronous API, we need to
+    // hold on to the "rpc" instance in order to get updates on the ongoing RPC.
```
Comment (vjpai) [inline examples/cpp/helloworld/greeter_async_client.cc]: Thanks.

**Comment by vjpai** [comment]
LGTM.

**Comment by vjpai** [comment]
I changed the title of the PR to better represent what is going on; hope you don't mind that.

**Pull request merged**